File: app/src/main/java/ua/com/foxminded/locationtrackera/model/locations/dao/LocationDatabase.java
package ua.com.foxminded.locationtrackera.model.locations.dao;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import ua.com.foxminded.locationtrackera.model.locations.UserLocation;

@Database(entities = UserLocation.class, version = 2, exportSchema = false)
public abstract class LocationDatabase extends RoomDatabase {

    private static LocationDatabase instance;

    public abstract UserLocationDao userLocationDao();

    public static synchronized LocationDatabase getInstance(Context context) {
        if (instance == null) {
            instance = Room.databaseBuilder(context.getApplicationContext(),
                    LocationDatabase.class, "location_database")
                    .fallbackToDestructiveMigration()
                    .build();
        }
        return instance;
    }
}

File: models_impl/src/main/java/ua/com/foxminded/locationtrackera/models_impl/locations/dao/LocationDatabase.kt
package ua.com.foxminded.locationtrackera.models_impl.locations.dao

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import ua.com.foxminded.locationtrackera.models.locations.UserLocation

@Database(entities = [UserLocation::class], version = 3, exportSchema = false)
abstract class LocationDatabase : RoomDatabase() {

    abstract fun userLocationDao(): UserLocationDao

    companion object {
        private var instance: LocationDatabase? = null

        @Synchronized
        fun getInstance(context: Context): LocationDatabase {
            return instance
                ?: Room.databaseBuilder(
                    context.applicationContext,
                    LocationDatabase::class.java,
                    "location_database"
                )
                    .fallbackToDestructiveMigration()
                    .build()
        }
    }
}